### PR TITLE
API: dtlike.astype(inty)

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -329,6 +329,7 @@ Other API changes
 - Default value of ``dtype`` in :func:`get_dummies` is changed to ``bool`` from ``uint8`` (:issue:`45848`)
 - :meth:`DataFrame.astype`, :meth:`Series.astype`, and :meth:`DatetimeIndex.astype` casting datetime64 data to any of "datetime64[s]", "datetime64[ms]", "datetime64[us]" will return an object with the given resolution instead of coercing back to "datetime64[ns]" (:issue:`48928`)
 - :meth:`DataFrame.astype`, :meth:`Series.astype`, and :meth:`DatetimeIndex.astype` casting timedelta64 data to any of "timedelta64[s]", "timedelta64[ms]", "timedelta64[us]" will return an object with the given resolution instead of coercing to "float64" dtype (:issue:`48963`)
+- :meth:`DatetimeIndex.astype`, :meth:`TimedeltaIndex.astype`, :meth:`PeriodIndex.astype` :meth:`Series.astype`, :meth:`DataFrame.astype` with ``datetime64``, ``timedelta64`` or :class:`PeriodDtype` dtypes no longer allow converting to integer dtypes other than "int64", do ``obj.astype('int64', copy=False).astype(dtype)`` instead (:issue:`??`)
```
Comment (jreback) [inline doc/source/whatsnew/v2.0.0.rst:332]: is this right? eg you can use copy=False to actually do this? how so

Comment (jbrockmendel) [reply doc/source/whatsnew/v2.0.0.rst:332]: passing copy=False here just avoids doing a double-copy, since the second .astype will always copy

Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:332]:
```suggestion
- :meth:`DatetimeIndex.astype`, :meth:`TimedeltaIndex.astype`, :meth:`PeriodIndex.astype` :meth:`Series.astype`, :meth:`DataFrame.astype` with ``datetime64``, ``timedelta64`` or :class:`PeriodDtype` dtypes no longer allow converting to integer dtypes other than "int64", do ``obj.astype('int64', copy=False).astype(dtype)`` instead (:issue:`49715`)
```

**Review by mroeschke — COMMENTED** [review]
Curious why an exception is only made for int64? (Assuming TimedeltaIndex.astype(int) is the nanos representation), `TimedeltaIndex(["1ns"])` should be able to fit in a smaller int type

**Comment by mroeschke** [comment]
Also a suggestion in a follow up PR: Could you collect all the "dtype conversion" whatsnew notes in the `Other API` section and give them their own dedicated section like under `Other API` -> `Dtype Consistency/Strictness`? I think it would be good to explain to users that pandas is trying to move toward dtype conversion operations that are less lossy and more explicit.

**Comment by mroeschke** [comment]
If I understand https://github.com/pandas-dev/pandas/issues/45034 correctly, it's generally concluded that it's okay to disallow casting ints except for int64? cc @jorisvandenbossche

**Comment by jorisvandenbossche** [comment]
> If I understand https://github.com/pandas-dev/pandas/issues/45034 correctly, it's generally concluded that it's okay to disallow casting ints except for int64?

That is not my reading of that issue .. I mentioned several times there that I personally don't see why such restriction is needed (while I also said that I don't think this is very important, though)

It's _also_ doing something different than what we actually said would change in the deprecation warnings (*"In a future version, this astype will return exactly the specified dtype instead of int64, and will raise if that conversion overflows."*)



> Continuing to allow int64 is a compromise, though in a perfect world I'd tell users to just `.view("i8")` instead.

In a perfect world, there would be no `view()` method on pandas objects (if you ask me ;))

**Comment by jbrockmendel** [comment]
> It's also doing something different than what we actually said would change in the deprecation warnings

yah, i reconsidered and changed my mind about that deprecation bc it is value-dependent behavior and in most cases just a very weird thing to .astype to and likely to be unintentional.

**Comment by mroeschke** [comment]
Personally I don't have a strong opinion on which direction to go as this astyping is strange. I could maybe interpret a user trying to do this to try to get an epoch timestamp/total seconds(?)/not sure what for Period. 

Agreed that changing this to achieve consistency between Series/Index is important though. I suppose I also like the stricter Series behavior to just signal that this astyping is ill defined

**Comment by jbrockmendel** [comment]
im now thinking the most likely case where a user would pass something other than int64 is if they pass `int` which on some builds gets interpreted as int32.  In this case they almost certainly wanted int64, and giving them int32 is going to throw them off (and make an unwanted copy).

we should disallow anything other than int64 (which is still more than pyarrow allows)

**Comment by jreback** [comment]
I am with @jbrockmendel here, I think we should *only* allow `int64`, this is super problematic with overflow and especially on windows.

**Comment by phofl** [comment]
I think int64 makes the most sense, int32 overflows real fast. I had similar issues to what @jbrockmendel mentioned while building a windows service in the past, this was super annoying. Only allowing int64 increases usability imo

**Comment by jbrockmendel** [comment]
@MarcoGorelli thoughts here?

**Comment by MarcoGorelli** [comment]
> @MarcoGorelli thoughts here?

No objections (no strong opinion either though, this isn't a topic I've delved into yet)

**Review by mroeschke — APPROVED** [review]
I think this is fine as is. Although it's slightly different from the original deprecation message the exception at least gives an alternative to convert to another int type

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**